deployment/src/main/java/io/quarkiverse/quinoa/deployment/packagemanager/PackageManagerInstall.java

```diff
@@ -70,14 +70,16 @@ private static Installation resolveInstalledNpmBinary(Path installDirectory) {
         final Path nodeBinPath = installDirectory.resolve(INSTALL_SUB_PATH)
                 .toAbsolutePath();
         final Path npmPath = installDirectory.resolve(NPM_PATH).toAbsolutePath();
-        final String platformNodeBinPath = convertToWindowsPathIfNeeded(nodeBinPath.toString());
-        final String platformNPMPath = convertToWindowsPathIfNeeded(npmPath.toString());
+        final String platformNodeBinPath = normalizePath(nodeBinPath.toString());
+        final String platformNPMPath = normalizePath(npmPath.toString());
         final String packageManagerBinary = NODE_BINARY + " " + platformNPMPath;
         return new Installation(platformNodeBinPath, packageManagerBinary);
     }
 
-    public static String convertToWindowsPathIfNeeded(String path) {
-        return PackageManager.isWindows() ? path.replaceAll("/", "\\\\") : path;
+    public static String normalizePath(String path) {
+        path = PackageManager.isWindows() ? path.replaceAll("/", "\\\\") : path;
+        path = path.contains(" ") ? "\"".concat(path).concat("\"") : path;
```

Comment:
@johnaohara shouldn't we do this only on Windows?

On other systems we might want to escape them with `\` instead?

Reply:
escaping the entire path with `"` should work in any env, I tested in bash/sh (linux)  and zsh (mac)